### DISCUSSION TITLE
complete replication of a dataset using zfs_autobackup?

### DISCUSSION THREAD

**Discussion by ahmgithubahm** [psy0rz/zfs_autobackup#270] [Q&A]
Hi all, and thanks @psy0rz for such a great tool!

In my previous homegrown ZFS backup script, I was able to completely replicate a data set, including all/any other random manual snapshots I'd taken on the source, to a target.  If I then subsequently deleted any of the random manual snaps on the source, they'd also disappear on the target at next backup.

Checking the script, I can see that initially, I'd take a snapshot THIS_BACKUP_SRC, and fully replicate it like this:

`zfs send --replicate "${THIS_BACKUP_SRC}" | zfs receive -Fu "${DST_POOL}/${SRC_FS}"`

For subsequent backups, I take another snapshot THIS_BACKUP_SRC and then do an incremental replication send from PREVIOUS_BACKUP_SRC onwards:

`zfs send --replicate -I "${PREVIOUS_BACKUP_SRC}" "${THIS_BACKUP_SRC}" | zfs receive -Fu "${DST_POOL}/${SRC_FS}"`

Is there any way that I can use zfs_autobackup like this?

Sorry if this is documented somewhere!

Thanks, Andy

**Comment by psy0rz** [comment]
yes, you can use `--other-snapshots` to do this:

https://github.com/psy0rz/zfs_autobackup/wiki/Manual#step-34-transferring-snapshots

**Comment by ahmgithubahm** [comment]
Thanks for reply @psy0rz .  Unfortunately, I don't think it's quite what I need, since I want these 'other snapshots' to disappear on the target when they disappear from the source, which the --other-snapshots option specifically does not do - "These other snapshots will never be destroyed".  I tested and as far as I can see, it's true - once an 'other snapshot' is sent to the target, your only option for later removal is to remove it manually.

**Reply by psy0rz** [reply]
Yeah im still not sure how we would implement this. We could make a --destroy-others that does this for example.

Or we could adjust the thinner so it also thins other snapshots on the target, by looking at the creation time.

But in both scenarios this would also destroy snapshots on the target that are created with perhaps another tool. But maybe thats ok if the admin chooses this.

**Reply by psy0rz** [reply]
Maybe other users here can respond with ideas for this.

**Reply by ahmgithubahm** [reply]
Thanks.

I'm very happy if the answer is just, "you can't do this with zfs_autobackup". 😄 zfs_autobackup is billed as a tool to make and replicate dataset snapshots, not one for replicating entire datasets exactly.

My little bash script doing the periodic full/incremental dataset replication works fine, with zfs_autobackup doing all the regular snapshotting/thinning on the source datasets.  I guess not everyone wants complete dataset replication, but are happy with periodic snapshots being replicated, something zfs_autobackup works perfectly for.  Thanks again.

**Reply by psy0rz** [reply]
Hehe no problem.

This has come up before and i was think about making some kind of zfs rsync, that just keeps stuff in sync. (Deleting stuff when needed)

See #114. Vote it up if its what you need :)
